numpy/lib/function_base.py

```diff
@@ -1912,10 +1911,16 @@ def diff(a, n=1, axis=-1):
     slice2[axis] = slice(None, -1)
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
+
+    if a.dtype == np.bool_:
+        da = a[slice1] ^ a[slice2]
```

Comment:
`!=` is kinda nice. Don't know if it add any funtionality. @eric-wieser Could you explain a bit more.